### PR TITLE
IEP-4: Kubernetes for hosting project applications

### PR DESCRIPTION
**Pull request by rtyler**
[Mailing list thread here](http://lists.jenkins-ci.org/pipermail/jenkins-infra/2016-December/000989.html)

### CODE REVIEW

**Comment by rtyler** [comment]
@olblak there are a few items in here we still need to figure out

**Review by olblak — COMMENTED** [review]
Kubernetes master -> As far as I know we can only specify agent size

File: iep-004/README.adoc [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+Kuberenetes clusters would be located in the "East US" Azure region.
+footnoteref:[regions,https://azure.microsoft.com/en-us/regions/]
+
+The virtual machines are all
+link:https://azure.microsoft.com/en-us/pricing/details/virtual-machines/series/#d-series[D-series]
+instances in order to provide an ideal balance of cost and performance. For
+more, see <<Costs>> below.
+
+
+*Kubernetes master*
```
Comment (olblak) [inline iep-004/README.adoc]: As far as I know we can only the number and the size of agents

File: iep-004/README.adoc [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+[source]
+----
++-------------------------+
+|  Kubernetes             |
+| |          ^            |
+| | +-----+  | Logs to k8s|
+| | | App +--`
+| | +-----+               |          +-----------------------+
+| | +---------+   Publish to blobs   |                       |
+| `>| Fluentd +--------------------->|   Azure Blob Storage  |
```
Comment (olblak) [inline iep-004/README.adoc]: Do we want to use the same storage account for logs than the one for k8s? 

In a second iteration, we may want to use a fluentd plugin to publish logs directly to OMS.
But right now it's easier to publish them to a shared storage.

File: iep-004/README.adoc [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+
+[NOTE]
+====
+This section is still a work in progress, awaiting and prototyping by
+link:https://github.com/olblak[Olivier Vernin]
+====
+
+== Motivation
+
+The motivation for centralizing container hosting is fairly
```
Comment (olblak) [inline iep-004/README.adoc]: It will also be easier to replicate containers hosting for different environments.

**Comment by olblak** [comment]
I created Jira Tasks INFRA-10[15-20]

File: iep-004/README.adoc (hunk 2 above)
Comment (rtyler) [reply iep-004/README.adoc]: I'm not sure I understand the question you're posing here.

If the k8s cluster requires its own storage account to operate properly, then yes, I believe that should be separate from the more general "logs" storage account

**Comment by olblak** [comment]
What would be the best way to update this PR?

**Comment by rtyler** [comment]
[Related pull request](https://github.com/jenkins-infra/jenkins-infra/pull/702) for orchestration of applications via Puppet

**Pull request merged**